02-peer-protocol.md

```diff
@@ -409,26 +409,49 @@ this channel will continue to use `option_static_remotekey` or `option_anchor_ou
 
 This message indicates that the funding transaction has reached the `minimum_depth` asked for in `accept_channel`. Once both nodes have sent this, the channel enters normal operating mode.
 
+As an extension, nodes which entirely funded the channel themselves,
+or has other reason to trust the peer, can send this message early, and
+optionally indicate the short_channel_id they will use to refer to it.
+
+
 1. type: 36 (`funding_locked`)
 2. data:
     * [`channel_id`:`channel_id`]
     * [`point`:`next_per_commitment_point`]
+    * [`funding_locked_tlvs`:`tlvs`]
+
+1. `tlv_stream`: `funding_locked_tlvs`
+2. types:
+    1. type: 1 (`short_channel_id`)
+    2. data:
+        * [`short_channel_id`:`short_channel_id`]
 
 #### Requirements
 
-The sender MUST:
-  - NOT send `funding_locked` unless outpoint of given by `funding_txid` and
+The sender:
+  - MUST NOT send `funding_locked` unless outpoint of given by `funding_txid` and
    `funding_output_index` in the `funding_created` message pays exactly `funding_satoshis` to the scriptpubkey specified in [BOLT #3](03-transactions.md#funding-transaction-output).
-  - wait until the funding transaction has reached `minimum_depth` before
-  sending this message.
-  - set `next_per_commitment_point` to the per-commitment point to be used
+  - MUST set `next_per_commitment_point` to the per-commitment point to be used
   for the following commitment transaction, derived as specified in
   [BOLT #3](03-transactions.md#per-commitment-secret-requirements).
+  - SHOULD set `short_channel_id`
+  - if it is the sole contributor to the funding transaction, or has reason to trust the peer:
+    - MAY send `funding_locked` before the funding transaction has reached `minimum_depth`
+	- MAY set `short_channel_id` to a fake value, if it will route payments to that `short_channel_id`.
```

Comment:
So the idea here is to leave the derivation of this value up to _both_ sides? Some implementations of zero conf channels I've seen in the wild, prescribe a schema (like using some bits from the pending chan ID which is selected by the funder as is anyway) so both sides can use the same value until things are confirmed on-chain, and a "real" scid is generated.

Reply:
On the other hand, the freedom allowed here to implementations is attractive, since one side is effectively just telling the other "I'm referring to that channel using this value until it confirms". The other direction can use another value as long as the receiver remembers that value (and doesn't even need to write it to disk?) when it sees it in an onion payload.